source/module_base/test/data/gen_ref.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from scipy.interpolate import CubicSpline
```

Comment:
> what's the usage of this python code?

This python script is used to systematically generate the reference data for the cross-check in unit test. In this PR, all the *.dat files in data/ are actually generated by this script. 

The unit test now consists of two sets of tests regarding the numerical reliability: 
- a set of checks with respect to the theoretical error bound
- a set of cross-checks with other implementation (which I chose scipy.interpolate.CubicSpline)

These two sets of tests are supposed to be arranged such that individual cases can be added/removed with minimal effort. Before this refactoring, reference data for cross-check are hard-coded (generated by some developer on his local machine and then copy-paste to unit test), which was very inconvenient and tedious. I felt it might be more appropriate to make this process automatic.

The script is provided for systematic generation of dat files used in cross-check; it has nothing to do with the main program of abacus. (so it is OK to have it removed)